engine/src/main/java/org/hibernate/validator/internal/util/TypeVariableBindings.java

```diff
@@ -69,7 +69,7 @@ else if ( genericSuperType instanceof ParameterizedType ) {
 
 				boolean typeParameterFoundInSubType = false;
 				for ( Entry<TypeVariable<?>, TypeVariable<?>> subTypeParameter : bindings.entrySet() ) {
-					if ( typeArgument.equals( subTypeParameter.getValue() ) ) {
+					if ( typeArgument == subTypeParameter.getValue() ) {
```

Comment:
OK, considering this part is not a hot spot anymore, I think we can go back to using equals. It made a big difference when `TypeVariableBindings` was called at runtime but it's probably negligible now that it's only a startup cost.

I don't see a case where `==` would fail but as it's deep into Java, let's be on the safe side.